lib/ext/patient.rb

```diff
@@ -138,6 +138,41 @@ def ethnicity
       end
     end
 
+    # Iterates through each data element to add
+    # 1. A Relevant DateTime if a Relevant Period is specified
+    # 2. A Relevant Period if a Relevant DateTime is specified
+    # This is used to normalize for eCQM calculations that may use one or the other
+    # The denormalize_as_datetime flag is used by the "denormalize_date_times" to return the record to the original state
+    def normalize_date_times
+      qdmPatient.dataElements.each do |de|
+        next unless de.respond_to?(:relevantDatetime) && de.respond_to?(:relevantPeriod)
```

Comment:
This checks that it responds to both relevantDatetime and relevantPeriod, but it doesn't check that exactly one is null/undefined. Do we have to worry about it ever having both be null or both be non-null/defined?